hera_cal/utils.py

```diff
@@ -157,94 +158,74 @@ def JD2LST(JD, longitude=21.42830):
     """
     Input:
     ------
-    JD : type=float or list of floats, julian date of observation
+    JD : type=float or list of floats containing Julian Date(s) of an observation
 
-    longitude : type=float, longitude of observer in degrees East, default=HERA Longitude
+    longitude : type=float, longitude of observer in degrees East, default=HERA longitude
 
     Output:
     -------
     Local Apparent Sidreal Time [radians]
-    """
-    if type(JD) is list or type(JD) is np.ndarray:
-        LST = []
-        for jd in JD:
-            t = Time(jd, format='jd', scale='utc')
-            LST.append(t.sidereal_time('apparent', longitude=longitude).value * np.pi / 12.0)
-        LST = np.array(LST)
-    else:
-        t = Time(JD, format='jd', scale='utc')
-        LST = t.sidereal_time('apparent', longitude=longitude).value * np.pi / 12.0
-
-    return LST
-
-
-def JD2RA(jd_array, longitude=21.42830):
-    """
-    Convert from julian date to RA at zenith, assuing J2000 epoch
-
-    jd_array : float or list of julian dates
 
-    lon : longitude of observer in degrees east
-    
-    return RA float (or array) in degrees
+    Notes:
+    ------
+    The Local Apparent Sidereal Time is *defined* as the right ascension in the current epoch.
     """
-    if type(jd_array) == list or type(jd_array) == np.ndarray:
+    # get JD type
+    if isinstance(JD, list) or isinstance(JD, np.ndarray):
         _array = True
     else:
         _array = False
-        jd_array = [jd_array]
-
-    # get observer
-    obs = ephem.Observer()
-    obs.epoch = ephem.J2000
-    obs.lon = longitude * np.pi / 180.0
+        JD = [JD]
 
-    # iterate over jd_array
-    RA = []
-    for JD in jd_array:
-        obs.date = Time(JD, format='jd', scale='utc').datetime
-        ra = obs.radec_of(0, np.pi/2)[0] * 180 / np.pi
-        RA.append(ra)
+    # iterate over JD
+    LST = []
+    for jd in JD:
+        # construct astropy Time object
+        t = Time(jd, format='jd', scale='utc')
+        # get LST in radians at epoch of jd
+        LST.append(t.sidereal_time('apparent', longitude=longitude).radian)
```

Comment:
While it's true that the longitude is treated as degrees in this function, I think it's better to make it an explicit `Quantity` object, e.g., define `lon = longitude * u.deg` (assuming you `import astropy.units as u`), and pass the new `lon` in as the keyword parameter. The docs only specify that a `Quantity`, `str`, or `None` is allowed for the `longitude` keyword, so support for it as a `float` may go away in the future. See http://docs.astropy.org/en/stable/api/astropy.time.Time.html#astropy.time.Time.sidereal_time.

Reply:
good call